lib/fs.js

```diff
@@ -2009,12 +2009,15 @@ function ReadStream(path, options) {
 
   if (this.start !== undefined) {
     if (typeof this.start !== 'number') {
-      throw new ERR_INVALID_ARG_TYPE('start', 'number', this.start);
+      throw new ERR_INVALID_ARG_TYPE('options.start', 'number', options.start);
+    }
+    if (!Number.isInteger(this.start)) {
+      throw new ERR_OUT_OF_RANGE('options.start', 'an integer', options.start);
     }
     if (this.end === undefined) {
       this.end = Infinity;
-    } else if (typeof this.end !== 'number') {
-      throw new ERR_INVALID_ARG_TYPE('end', 'number', this.end);
+    } else if (!Number.isInteger(this.end)) {
+      throw new ERR_INVALID_ARG_TYPE('end', 'integer', this.end);
```

Comment:
The same as above: distinguishing the type from the rest of the error would be good. You can actually copy the same part and just change what variable it applies to.

Reply:
For example, change
```js
    if (this.end === undefined) {
      this.end = Infinity;
    } else if (!Number.isInteger(this.end)) {
      throw new ERR_INVALID_ARG_TYPE('end', 'integer', this.end);
    }
```
to
```js
    if (this.end === undefined) {
      this.end = Infinity;
    }else{
      if(typeof this.end!=='number'){
        throw new ERR_INVALID_ARG_TYPE('end','number',this.end)
      }
      if (!Number.isInteger(this.end)) {
        throw new ERR_OUT_OF_RANGE('end', 'integer', this.end);
      }
    }
```
.